be/src/exec/pipeline/scan/chunk_buffer_limiter.h

```diff
@@ -0,0 +1,71 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+#include <mutex>
+
+namespace starrocks::pipeline {
+
+class ChunkBufferLimiter;
+using ChunkBufferLimiterPtr = std::unique_ptr<ChunkBufferLimiter>;
+
+class ChunkBufferLimiter {
```

Comment:
should add comment for this class to explain when and how will it be used.

Reply:
need `virtual ~ChunkBufferLimiter`

Reply:
seems could be a common thresholder or limiter?
For example for disk or network throughput.

And you need to explain if this class interface is thread-safe.